mirdata/gtzan_genre.py

```diff
@@ -0,0 +1,154 @@
+# -*- coding: utf-8 -*-
+"""GTZAN-Genre Dataset Loader
+
+
+"""
+
+from __future__ import absolute_import
+from __future__ import division
+from __future__ import print_function
+
+import os
+import librosa
+
+from mirdata import download_utils
+from mirdata import utils
+
+
+DATASET_DIR = "GTZAN-Genre"
+
+DATASET_REMOTE = download_utils.RemoteFileMetadata(
+    filename="genres.tar.gz",
+    url="http://opihi.cs.uvic.ca/sound/genres.tar.gz",
+    checksum="5b3d6dddb579ab49814ab86dba69e7c7",
+    destination_dir="gtzan_genre",
+)
+
+
+DATA = utils.LargeData("gtzan_genre_index.json")
+
+
+class Track(object):
+    """GTZAN-Genre track class
+
+    Args:
+        track_id (str): track id of the track
+        data_home (str): Local path where the dataset is stored.
+            If `None`, looks for the data in the default directory, `~/mir_datasets/GuitarSet`
+
+    Attributes:
+        track_id (str): track id
+        genre (str): annotated genre
+        audio_path (str): relative audio path (e.g. "classical/classical.00000.wav")
+        abs_audio_path (str): absolute audio path
+    """
+
+    def __init__(self, track_id, data_home=None):
+        if track_id not in DATA.index:
+            raise ValueError(
+                "{} is not a valid track ID in GTZAN-Genre".format(track_id)
+            )
+
+        self.track_id = track_id
+
+        if data_home is None:
+            data_home = utils.get_default_dataset_path(DATASET_DIR)
+
+        self._data_home = data_home
+        self._track_paths = DATA.index[track_id]
+
+        self.genre = track_id.split(".")[0]
+        self.audio_path = self._track_paths["audio"][0]
```

Comment:
I don't think it's useful to have the relative path exposed explicitly in an attribute - can you merge it into one line:
`self.audio_path = os.path.join(self._data_home, self._track_paths["audio"][0]`